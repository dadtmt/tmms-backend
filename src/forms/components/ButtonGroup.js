import R from 'ramda'
import React, { PropTypes } from 'react'
import { Button, FormGroup, Glyphicon } from 'react-bootstrap'

const ButtonGroup = props => (
  <FormGroup>
    <Button {...R.omit('glyph')(props)} >
      <Glyphicon glyph={props.glyph} />
    </Button >
  </FormGroup>
)

ButtonGroup.propTypes = {
  glyph: PropTypes.string.isRequired
}

export default ButtonGroup
