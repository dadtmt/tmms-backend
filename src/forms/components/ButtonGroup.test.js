import React from 'react'
import renderer from 'react-test-renderer'

import ButtonGroup from './ButtonGroup'

it('should render correctly', () => {
  const props = {
    glyph: 'plus',
    onClick: jest.fn(),
    title: 'add'
  }
  expect(
    renderer.create(<ButtonGroup {...props} />).toJSON()
  ).toMatchSnapshot()
})
