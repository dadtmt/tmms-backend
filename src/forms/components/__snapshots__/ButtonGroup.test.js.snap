exports[`test should render correctly 1`] = `
<div
  className="form-group">
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    title="add"
    type="button">
    <span
      className="glyphicon glyphicon-plus" />
  </button>
</div>
`;
